apps/massa_proxy/lib/massa_proxy/runtime/grpc/protocol/action/stream_handler.ex

```diff
@@ -3,58 +3,49 @@ defmodule MassaProxy.Runtime.Grpc.Protocol.Action.Stream.Handler do
   This module is responsible for handling stream requests of the Action protocol
   """
 
-  alias Google.Protobuf.Any
-  alias Cloudstate.{Action.ActionCommand, Metadata}
   alias Cloudstate.Action.ActionProtocol.Stub, as: ActionClient
+  alias MassaProxy.Runtime.Grpc.Protocol.Action.Protocol, as: ActionProtocol
 
-  import MassaProxy.Util, only: [get_connection: 0, get_type_url: 1]
+  import MassaProxy.Util, only: [get_connection: 0]
 
-  def handle_streamed(%{stream: stream, message: message} = payload) do
-    messages = build_stream(message, payload)
+  def handle_streamed(%{stream: stream} = ctx) do
+    messages = ActionProtocol.build_stream(ctx)
 
     with {:ok, conn} <- get_connection(),
          client_stream = ActionClient.handle_streamed(conn),
          :ok <- client_stream |> run_stream(messages) |> Stream.run(),
          {:ok, consumer_stream} <- GRPC.Stub.recv(client_stream) do
       consumer_stream
-      |> Stream.each(fn {:ok, r} -> GRPC.Server.send_reply(stream, unpack_response(r)) end)
+      |> Stream.each(fn {:ok, r} ->
+        GRPC.Server.send_reply(stream, ActionProtocol.decode(ctx, r))
+      end)
       |> Stream.run()
     else
       {:error, _reason} = err -> err
```

Comment:
You don't have to worry about that right now but I think we can do better here. We need to think about what would be the best mistake to generate an adequate response for the client in cases of failure.